src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -1159,10 +1167,14 @@ public SolrInputDocuments toSolrDocs(IndexableDataset indexableDataset, Set<Long
                     if(end!=null) {
                         datafileSolrInputDocument.addField(SearchFields.EMBARGO_END_DATE, end.toEpochDay()); 
                     }
-                    
+                    if(start!=null) {
+                        datafileSolrInputDocument.addField(SearchFields.RETENTION_END_DATE, start.toEpochDay());
+                    }
                     /* Full-text indexing using Apache Tika */
                     if (doFullTextIndexing) {
-                        if (!dataset.isHarvested() && !fileMetadata.getDataFile().isRestricted() && !fileMetadata.getDataFile().isFilePackage()) {
+                        if (!dataset.isHarvested() && !fileMetadata.getDataFile().isRestricted()
+                                && !fileMetadata.getDataFile().isFilePackage()
+                                && fileMetadata.getDataFile().getRetention() == null) {
```

Comment:
This seems somewhat overprotective, especially if we're talking about 10 year + retention periods. In most cases, reindexing is likely to occur fairly soon after. Without inventing a new event trigger mechanism I could see a couple options to try handling this:
* an optional flag to allow fulltext indexing for unexpired files
* a check of the last indexed date whenever  isRetentionExpired is checked - if it hasn't been indexed after the expire date, add it for indexing. This only slows things when expired files are involved and just involves one additional date check. It does mean that the file would remain full-text indexed until the first hit
* add an API call to reindex newly expired files that could be called by a cron script.

Is any of this worth doing now?

Reply:
I am extremely cautious when there are legal issues involved, so I would not go for the second point, because it seems not `waterproof`. 

The optional flag together with the added API endpoint would be a good thing for a follow-up PR, but not right now.